### PR TITLE
add github actions

### LINKED ISSUE #3233: Reminder: Transition to travis.com, use travis github app + checks api
https://blog.github.com/2018-05-07-introducing-checks-api/
https://blog.travis-ci.com/2018-05-07-announcing-support-for-github-checks-api-on-travis-ci-com
https://blog.travis-ci.com/2018-05-02-open-source-projects-on-travis-ci-com-with-github-apps?utm_source=Broadcast&utm_campaign=2may_release

https://docs.travis-ci.com/user/open-source-on-travis-ci-com/#Existing-Open-Source-Repositories-on-travis-ci.org
https://docs.travis-ci.com/user/migrate/open-source-repository-migration#migrating-a-repository

https://developer.github.com/changes/2018-05-07-new-checks-api-public-beta/
https://developer.github.com/v3/checks/

**Update:
https://docs.travis-ci.com/user/migrate/open-source-on-travis-ci-com/#existing-open-source-repositories-on-travis-ciorg**

### LINKED ISSUE #4170: Cockatrice on macOS 11 Big Sur
Currently we do not provide any releases for macOS Big Sur, previous releases are not compatible.

As a temporary measure you can download a **experimental** build from the artifacts on this pr here:
https://github.com/Cockatrice/Cockatrice/actions/runs/371281906

You will need to be logged in to GitHub to see the downloads!
You can then click the link to download the BigSur one and unzip it in order to install.

If you find any bugs in this experimental version be sure to report them to us as soon as possible! We'll be able to fix them before we make a release!

### PR DESCRIPTION
**Pull request by ebbit1q**
## Related Ticket(s)
- Fixes #3233
- Fixes #4170
- replaces https://github.com/Cockatrice/Cockatrice/pull/4103 https://github.com/Cockatrice/Cockatrice/pull/4077 https://github.com/Cockatrice/Cockatrice/pull/4159

## Short roundup of the initial problem
Travis CI doesn't want to support oss projects anymore, we will use github's own ci instead for mac and linux builds now. It even seems to be faster even though it was a pain to set up.
There is also an option to do the same on windows if we figure out if we (want to) get 32 bit versions too.

## What will change with this Pull Request?
- friendship ended with travis
- github actions is new best friend
- you can see my test release on my fork https://github.com/ebbit1q/Cockatrice/releases/tag/TEST_ACTIONS
- the interface is at https://github.com/ebbit1q/Cockatrice/actions with the logs

## notes:
- I made a build for two different versions of xcode, if this correlates to mac versions like on travis is yet to be seen.
- ccache on macOS does not work for some reason, this was the same for travis last I checked
- testing on macOS does not work for some more obscure reason https://github.com/ebbit1q/Cockatrice/runs/1375443229?check_suite_focus=true#step:7:831
- the mysql connector is not found on macOS https://github.com/ebbit1q/Cockatrice/runs/1382891815?check_suite_focus=true#step:8:37
- linux builds take 2 minutes installing dependencies and 1 minute building, if github provided more cache space we could cache these too, but we only have 5gb and each image takes 2
- ccache's configuration is by default a maximum of 5gb for each image, we'll have to consider if this is an issue or not. The most this should use is around 700mb in my opinion but we'll have to see if it's the same for mac if we get that working. (we can tweak this later, consider we don't think about this at all on travis either #3781)

### CODE REVIEW

**Comment by ebbit1q** [comment]
Do you want me to revamp the naming scheme? currently all names have to be changed to include the release "name" eg Cockatrice-Blessed_Sanctuary-2.7.5-DebianBuster.deb (currently outputs Cockatrice-2.7.6-DebianBuster.deb instead)

**Comment by ebbit1q** [comment]
the github actions that are triggered by the pull request are visible on the "checks" tab btw

**Comment by ebbit1q** [comment]
big sur requires cmake 3.19 https://gitlab.kitware.com/cmake/cmake/-/issues/20889

File: .github/workflows/linux-builds.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          - distro: ArchLinux
+            package: skip # we are packaged in arch already
+            allow-failure: yes
+
+          - distro: DebianBuster
+            package: DEB
+
+          - distro: Fedora33
+            package: RPM
+            test: skip # Fedora is our slowest build
```
Comment (ZeldaZach) [inline .github/workflows/linux-builds.yml:39]: We _might_ want to change this in the future, but for now I'm okay with skipping it

File: .github/workflows/macos-builds.yml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+      - name: Package for ${{matrix.target}}
+        if: matrix.make_package
+        id: build_package
+        shell: bash
+        working-directory: build
+        run: |
+          # temporary workaround for big sur images having old cmake
+          if [[ ${{matrix.os}} == macos-11.0 ]]; then
+            curl -L https://github.com/Kitware/CMake/releases/download/v3.19.0/cmake-3.19.0-Darwin-x86_64.tar.gz | tar -xz
```
Comment (ZeldaZach) [inline .github/workflows/macos-builds.yml:145]: to be removed in the future when cmake 3.19 comes around

**Review by ZeldaZach — APPROVED** [review]
Lets merge this in and see where we're at

**Pull request merged**